src/Omega_h_array.hpp

```diff
@@ -47,61 +65,20 @@ class Write {
   Write(LO size_in, T offset, T stride, std::string const& name = "");
   Write(std::initializer_list<T> l, std::string const& name = "");
   Write(HostWrite<T> host_write);
-  OMEGA_H_INLINE LO size() const OMEGA_H_NOEXCEPT {
-#ifdef OMEGA_H_CHECK_BOUNDS
-    OMEGA_H_CHECK(exists());
-#endif
-#ifdef OMEGA_H_USE_KOKKOS
-    return static_cast<LO>(view_.size());
-#else
-    return static_cast<LO>(shared_alloc_.size() / sizeof(T));
-#endif
-  }
-  OMEGA_H_DEVICE T& operator[](LO i) const OMEGA_H_NOEXCEPT {
-#ifdef OMEGA_H_CHECK_BOUNDS
-    OMEGA_H_CHECK_OP(0, <=, i);
-    OMEGA_H_CHECK_OP(i, <, size());
-#endif
-#ifdef OMEGA_H_USE_KOKKOS
-    return view_(i);
-#else
-    return data()[i];
-#endif
-  }
-  OMEGA_H_INLINE T* data() const noexcept {
-#ifdef OMEGA_H_USE_KOKKOS
-    return view_.data();
-#else
-    return static_cast<T*>(shared_alloc_.data());
-#endif
-  }
+  OMEGA_H_INLINE LO size() const OMEGA_H_NOEXCEPT;
+  OMEGA_H_DEVICE T& operator[](LO i) const OMEGA_H_NOEXCEPT;
+  OMEGA_H_INLINE T* data() const noexcept;
 #ifdef OMEGA_H_USE_KOKKOS
   OMEGA_H_INLINE Kokkos::View<T*> const& view() const { return view_; }
 #endif
   void set(LO i, T value) const;
   T get(LO i) const;
-#ifdef OMEGA_H_USE_KOKKOS
-  OMEGA_H_INLINE long use_count() const { return view_.use_count(); }
-#else
-  inline int use_count() const { return shared_alloc_.alloc->use_count; }
-#endif
-  OMEGA_H_INLINE bool exists() const noexcept {
-#if defined(OMEGA_H_USE_KOKKOS)
-    return view().data() != nullptr
-#if defined(KOKKOS_ENABLE_DEPRECATED_CODE) && (!defined(__CUDA_ARCH__))
-           /* deprecated Kokkos behavior: zero-span views have data()==nullptr
-            */
-           || view().use_count() != 0
-#endif
-        ;
-#else
-    return shared_alloc_.data() != nullptr;
-#endif
-  }
+  OMEGA_H_INLINE long use_count() const;
+  OMEGA_H_INLINE bool exists() const noexcept;
 #ifdef OMEGA_H_USE_KOKKOS
   std::string name() const;
 #else
-  std::string const& name() const;
+      std::string const& name() const;
```

Comment:
whitespace

Reply:
Thank you. Fixed with [8f350da](https://github.com/SCOREC/omega_h/pull/60/commits/8f350da36549a245e0b3c26c699c7db2eaf1ff6d).